modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -375,8 +582,19 @@ def _launch_tasks(cls, splits: list, **partition_kwargs) -> Tuple[list, list, li
         partition_ids = []
         index_ids = []
         dtypes_ids = []
-        for start, end in splits:
-            partition_kwargs.update({"start": start, "end": end})
+        for split_data in splits:
+            rows_considered = split_data[3]
+            # rows_considered == 0 means that partition will contain empty DataFrame
+            # and in this case task deployment can be bypassed
+            if rows_considered == 0:
+                continue
```

Comment:
It seems to me that this does not significantly save time. 

Is `rows_considered` var being used for something else?

Reply:
Launching of tasks which will output empty frames bypassed here to avoid further processing these frames metadata (was discussed in [comment](https://github.com/modin-project/modin/pull/2607#discussion_r574306355)). `rows_considered` var  is used mostly inside `_read_rows` and `partitioned_file` functions.

Reply:
We can do this check in place where `splits` created - in `partitioned_file`. Then we localize the work with `rows_considered` variable in `partitioned_file` function.

Reply:
Yes, i agree, suggestion was applied.